blueflood-http/src/integration-test/java/com/rackspacecloud/blueflood/outputs/handlers/HttpEnumRollupIntegrationTest.java

```diff
@@ -0,0 +1,131 @@
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package com.rackspacecloud.blueflood.outputs.handlers;
+
+import com.rackspacecloud.blueflood.cache.MetadataCache;
+import com.rackspacecloud.blueflood.http.HttpClientVendor;
+import com.rackspacecloud.blueflood.io.*;
+import com.rackspacecloud.blueflood.rollup.Granularity;
+import com.rackspacecloud.blueflood.service.*;
+import com.rackspacecloud.blueflood.types.*;
+import org.apache.http.impl.client.DefaultHttpClient;
+import org.junit.*;
+
+import java.util.*;
+
+public class HttpEnumRollupIntegrationTest extends IntegrationTestBase {
+    // A timestamp 2 days ago
+    private final long baseMillis = Calendar.getInstance().getTimeInMillis() - 172800000;
+    private final String tenantId = "ac" + IntegrationTestBase.randString(8);
+    private final String enumMetricName = "enumMet_"+IntegrationTestBase.randString(8);
+
+    List<Locator> enumLocators;
+    private static int queryPort = 20000;
+    private static HttpQueryService httpQueryService;
+    private static HttpClientVendor vendor;
+    private static DefaultHttpClient client;
+
+    private HttpRollupsQueryHandler httpHandler;
+    private final Map<Locator, Map<Granularity, Integer>> enumlocatorToPoints = new HashMap<Locator, Map<Granularity,Integer>>();
+
+
+    @BeforeClass
+    public static void setUpHttp() {
+        queryPort = Configuration.getInstance().getIntegerProperty(HttpConfig.HTTP_METRIC_DATA_QUERY_PORT);
+        httpQueryService = new HttpQueryService();
+        httpQueryService.startService();
+        vendor = new HttpClientVendor();
+        client = vendor.getClient();
+    }
+
+    @Before
+    public void setUp() throws Exception {
+        super.setUp();
+        AstyanaxWriter writer = AstyanaxWriter.getInstance();
+
+        enumLocators = new ArrayList<Locator>();
+        for(int i=0; i< 1440; i++) {
+            final long curMillis = baseMillis + ( i* 60000);
+            final List<IMetric> metrics = new ArrayList<IMetric>();
+            final IMetric enumMetric = getEnumMetric(enumMetricName, tenantId, curMillis);
+            metrics.add(enumMetric);
+            enumLocators.add(enumMetric.getLocator());
+
+            MetadataCache.getInstance().put(enumMetric.getLocator(), MetricMetadata.TYPE.name().toLowerCase(), null);
+            MetadataCache.getInstance().put(enumMetric.getLocator(), MetricMetadata.ROLLUP_TYPE.name().toLowerCase(), RollupType.ENUM.toString());
+            writer.insertMetrics(metrics, CassandraModel.CF_METRICS_PREAGGREGATED_FULL);
+        }
+
+        httpHandler = new HttpRollupsQueryHandler();
+
+        // generate every level of rollup for the raw data
+        Granularity g = Granularity.FULL;
+        while (g != Granularity.MIN_1440) {
+            g = g.coarser();
```

Comment:
@VinnyQ Yes that's the correct thing to do, because we are specifying the destination granularity. The full resolution data already exists.